Examples/Examples/All Examples/SceneKitExample.swift

```diff
@@ -0,0 +1,228 @@
+import UIKit
+import SceneKit
+import MapboxMaps
+
+@objc(SceneKitExample)
+
+public class SceneKitExample: UIViewController, ExampleProtocol, CustomLayerHost {
+
+    internal var mapView: MapView!
+    public var peer: MBXPeerWrapper?
+
+    let modelOrigin = CLLocationCoordinate2D(latitude: -35.39847, longitude: 148.9819)
+    var renderer: SCNRenderer!
+    var scene: SCNScene!
+    var modelNode: SCNNode!
+    var cameraNode: SCNNode!
+    var textNode: SCNNode!
+    var useCPUOcclusion = false
+
+    override public func viewDidLoad() {
+        super.viewDidLoad()
+
+        self.mapView = MapView(with: view.bounds, resourceOptions: resourceOptions())
+        mapView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+        self.view.addSubview(mapView)
+        mapView.cameraManager.setCamera(
+            centerCoordinate: self.modelOrigin,
+                        zoom: 18,
+                     bearing: 180,
+                       pitch: 60
+        )
+        self.mapView.update { (mapOptions) in
+            mapOptions.render.presentsWithTransaction = true
+        }
+
+        self.mapView.on(.styleLoadingFinished) { [weak self] _ in
+            self?.addModelAndTerrain()
+        }
+    }
+
+    override public func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+    }
+
+    func addModelAndTerrain() {
+        try! mapView.__map.addStyleCustomLayer(forLayerId: "Custom",
+                                                layerHost: self,
+                                            layerPosition: LayerPosition(above: nil, below: "waterway-label", at: nil))
+
+        var demSource = RasterDemSource()
+        demSource.url = "mapbox://mapbox.mapbox-terrain-dem-v1"
+        demSource.tileSize = 514
```

Comment:
Is 514 correct?

Reply:
514 is used for padded (border ready) DEM tiles. I wanted to test it here and left the value.

Padded DEM tiles are performance improvement, especially for hillshade: borders are prepared on server side and there is no need to [backfill borders from neighboring DEM tiles](https://github.com/mapbox/mapbox-gl-js/blob/4d32dbe61128d279821f7f50dc0c1d9d804b7ed4/src/data/dem_data.js#L53). 

Anyway, it is unnecessary, additional information for this PR. I'll remove before merging.